### ISSUE TITLE
Investigate translation bug with nested floating renderables

### ISSUE THREAD

**Issue by parasyte** [melonjs/melonJS#649]
This issue pops up all the time on the forum/in chat. A floating Renderable inside a floating Container will move with the viewport in the opposite direction (it's weird). This is known to happen in 2.0.x. Ticket is to track its status for 2.1

**Comment by obiot** [comment]
does this still happen with all the changes made by Aaron ?

**Comment by agmcleod** [comment]
Yes. It doesn't handle the floating state very well.

**Comment by parasyte** [comment]
I forgot to link the forum post in question: https://groups.google.com/d/msg/melonjs/AFqhE1u3G2U/g9cnV7eTaRsJ

**Comment by agmcleod** [comment]
2504f2e

Added a version of the updateAbsoluteBounds() to the container that invokes the resize method. It still uses a child bounds object, as I figure it's better than constructing a new Rect each time :). No longer runs on add/remove child calls.

Not sure what you mean by clipping regions in the commit, but this is a bit of a simpler version.

**Comment by parasyte** [comment]
Canvas clipping regions:

```javascript
var context = renderer.getContext();
context.save();
context.clip(this.pos.x, this.pos.y, this.width, this.height);
// ... loop through children here and call their draw method
context.restore();
```

https://developer.mozilla.org/samples/canvas-tutorial/6_2_canvas_clipping.html

For WebGL, use the stencil buffer.

**Comment by agmcleod** [comment]
Wouldn't this have the same problem of drawing objects outside the container dimensions? Or if we're not worried about that, can even just keep the inViewport test against the container itself, and not worry about updating the rect.

**Comment by aaschmitz** [comment]
@agmcleod with the latest adjustments, the particles examples show the error below in console:

TypeError: this.getBounds(...) is undefined melonJS-2.1.0.js:8213

**Comment by agmcleod** [comment]
Yeah i didn't do any changes to that one yet. It's probably still broken in webgl for the regular debugger too. Was just making some fixes.

**Comment by parasyte** [comment]
@agmcleod My argument is that it would simplify the melonJS code to clip all children that are outside of their container bounds, instead of attempting to iterate all children recursively to compute the minimum bounding box and resize it on every frame.

**Comment by obiot** [comment]
same here, except someone is super motivated to fix within the next couple of days ?

**Comment by parasyte** [comment]
IMHO, this should be moved to the next major milestone (3.0.0). Opinions?

**Comment by obiot** [comment]
Not sure when 3.0.0 would happen, but i would agree on at least 2.2.x

**Comment by parasyte** [comment]
My suggestion for 3.0.0 is based on the impact of changing the container draw to clipping its children: https://github.com/melonjs/melonJS/issues/649#issuecomment-75360960

And anyway... we could release 3.0.0 tomorrow if we really wanted. :laughing:

**Comment by parasyte** [comment]
Verified fixed in 2.1.x:

* 2.0.2: http://jsfiddle.net/parasyte/oceabx5j/1/
* 2.1.1: http://jsfiddle.net/parasyte/oceabx5j/

I'll create a new ticket for the clipping.
